### ISSUE TITLE
Insufficient parameters error

### ISSUE THREAD

**Issue by WesThorburn** [yajra/laravel-datatables#89]
I'm using laravel-datatables for the first time. I believe I have installed it correctly. 

I'm getting a fatal "Insufficient parameters" error. (Exception in Datatables.php line 69:)

Here is my line of code:
``'stockDt' => Datatables::of(Stock::all())->make(true),``

There are at least two other users over on the Laracasts forum reporting the same issue. (https://laracasts.com/discuss/channels/eloquent/yajra-laravel-datatables?page=1)

Are we doing something wrong here? Or is there a problem?

**Comment by yajra** [comment]
this means that you are calling the url directly. You need to pass the required parameters like `draw=1&columns....`. These parameters are automatically added on javascript.

**Comment by WesThorburn** [comment]
I don't understand what you mean by that.

Where about's do I pass the required parameters exactly? If you have an example you can show me, that would be great.

Also, I should add that this was the example code I'm using as a reference. The way I'm getting my Eloquent collection is slightly differently obviously. 
 
```
public function getData()
{
    $users = $this->users->select('*');
    return Datatables::of($users)
        ->addColumn('action', 'action here')
        ->make(true);
}
```

**Comment by yajra** [comment]
what I mean is are you trying to access it directly in the browser like `http://laravel.app/users/data`? If yes, then you should add the parameters like 'http://laravel.app/users/data?draw=1&columns....`. Note that the plugin must work side-by-side with DataTables jQuery plugin. See the demo app I created for ref http://datatables.yajrabox.com/

**Comment by WesThorburn** [comment]
I have the DataTables jQuery plugin working just fine, no issues there.

I'm not passing any parameters in through the browser at all. The page i'm trying to load this on is my site's index page, so on my local machine that's simply 'localhost:8000.' However, I don't think that's the issue I'm having.

The error is being thrown from the controller. As soon as I call the Datatables:: facade and pass my eloquent collection in as the parameter, it fails with the "Insufficient parameters" error. It fails with that same error regardless whether or not I'm returning a view.

**Comment by vmadev** [comment]
Don't try to call the API because you will get that exact error. Simply call your view route... Ajax request to the api route are handled behind the scenes.

**Comment by adavidromero** [comment]
I had this same error, in my case the problem was in the javascript, I wrote the parameter serverside:true instead of serverSide:true fixed that and then it worked!

**Comment by WesThorburn** [comment]
Okay, I fixed it. 

The error was being thrown when I was trying to return the Datatables instance with my other view variables, it doesn't work that way.

To fix it, I created a separate function in my controller that is solely responsible for returning the Datatables instance. Then, I set up a route in my routes file that points to the function. The Javascript in the view that initializes the table looks something like this:
```
$(document).ready(function() {
    $('#example').dataTable( {
        "processing": true,
        "serverSide": true,
        "ajax": "route/to/datatables/function"
    } );
} );
```
I'm not the first person to make this mistake, and I know I won't be the last. Hopefully this will help someone else having the same issue.
